Multiple definition of `g_platform` when linking libvariorum.so

Exists on latest release tag and dev branch. Looking into it myself, but maybe you already know the answer. Include guard is present in the named file, multifile grep only turns up one definition. Can't find any syntax errors leading to an accidental redefinition.

```
[ 19%] Linking Fortran shared library libvariorum.so
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: CMakeFiles/variorum.dir/variorum.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: multiple definition of `g_platform'; CMakeFiles/variorum.dir/config_architecture.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: CMakeFiles/variorum.dir/Intel/config_intel.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: multiple definition of `g_platform'; CMakeFiles/variorum.dir/config_architecture.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: Intel/CMakeFiles/variorum_intel.dir/clocks_features.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: multiple definition of `g_platform'; CMakeFiles/variorum.dir/config_architecture.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: Intel/CMakeFiles/variorum_intel.dir/counters_features.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: multiple definition of `g_platform'; CMakeFiles/variorum.dir/config_architecture.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: Intel/CMakeFiles/variorum_intel.dir/power_features.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: multiple definition of `g_platform'; CMakeFiles/variorum.dir/config_architecture.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: Intel/CMakeFiles/variorum_intel.dir/thermal_features.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: multiple definition of `g_platform'; CMakeFiles/variorum.dir/config_architecture.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: Intel/CMakeFiles/variorum_intel.dir/misc_features.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: multiple definition of `g_platform'; CMakeFiles/variorum.dir/config_architecture.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: Intel/CMakeFiles/variorum_intel.dir/SandyBridge_2A.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: multiple definition of `g_platform'; CMakeFiles/variorum.dir/config_architecture.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: Intel/CMakeFiles/variorum_intel.dir/SandyBridge_2D.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: multiple definition of `g_platform'; CMakeFiles/variorum.dir/config_architecture.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: Intel/CMakeFiles/variorum_intel.dir/IvyBridge_3E.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: multiple definition of `g_platform'; CMakeFiles/variorum.dir/config_architecture.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: Intel/CMakeFiles/variorum_intel.dir/Haswell_3F.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: multiple definition of `g_platform'; CMakeFiles/variorum.dir/config_architecture.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: Intel/CMakeFiles/variorum_intel.dir/Broadwell_4F.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: multiple definition of `g_platform'; CMakeFiles/variorum.dir/config_architecture.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: Intel/CMakeFiles/variorum_intel.dir/Skylake_55.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: multiple definition of `g_platform'; CMakeFiles/variorum.dir/config_architecture.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: Intel/CMakeFiles/variorum_intel.dir/KabyLake_9E.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: multiple definition of `g_platform'; CMakeFiles/variorum.dir/config_architecture.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: Intel/CMakeFiles/variorum_intel.dir/msr_core.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: multiple definition of `g_platform'; CMakeFiles/variorum.dir/config_architecture.c.o:/home/R0b0t1/devel/variorum/src/variorum/config_architecture.h:252: first defined here
collect2: error: ld returned 1 exit status
make[2]: *** [variorum/CMakeFiles/variorum.dir/build.make:193: variorum/libvariorum.so] Error 1
make[1]: *** [CMakeFiles/Makefile2:605: variorum/CMakeFiles/variorum.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

--- a/src/variorum/config_architecture.c
+++ b/src/variorum/config_architecture.c
@@ -31,6 +31,8 @@
 #include <config_arm.h>
 #endif
 
+struct platform g_platform;
+
 #ifdef VARIORUM_LOG
 int variorum_enter(const char *filename, const char *func_name, int line_num)
 #else

--- a/src/variorum/config_architecture.h
+++ b/src/variorum/config_architecture.h
@@ -249,7 +249,7 @@ struct platform
 //    void (*set_performance_counters)();
 #endif
 
-struct platform g_platform;
+extern struct platform g_platform;
 
 #ifdef VARIORUM_LOG
 int variorum_enter(const char *filename,

